### PR TITLE
Add CloudProfile validator

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind enhancement

Similar to https://github.com/gardener/gardener-extension-provider-aws/pull/250

**Release note**:

```improvement operator
Cloud specific settings of Alicloud `CloudProfiles` are now being validated.
```
```action operator
The `gardener-extension-validator-alicloud` Helm chart as well as different assets inside have been renamed to the more general term `gardener-extension-admission-alicloud`. Please consider to take corresponding action if you don't use Helm to manage your deployment in the Garden cluster.
```
```action operator
The Docker image `eu.gcr.io/gardener-project/gardener/extensions/validator-alicloud` will no longer be maintained as of this release in favor of the successor `eu.gcr.io/gardener-project/gardener/extensions/admission-alicloud`. Please consider replacing any references to the image `eu.gcr.io/gardener-project/gardener/extensions/validator-alicloud` by `eu.gcr.io/gardener-project/gardener/extensions/admission-alicloud`.
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,8 +15,14 @@ COPY charts /charts
...
 ENTRYPOINT ["/gardener-extension-provider-alicloud"]
 
+############# gardener-extension-admission-alicloud
+FROM base as gardener-extension-admission-alicloud
+
+COPY --from=builder /go/bin/gardener-extension-admission-alicloud /gardener-extension-admission-alicloud
+ENTRYPOINT ["/gardener-extension-admission-alicloud"]
+
 ############# gardener-extension-validator-alicloud
 FROM base AS gardener-extension-validator-alicloud
```
Comment (minchaow) [inline Dockerfile:25]: I see you renaming the `validator` to `admission`, why here still need to build `validator` image?

**Review by minchaow — APPROVED** [review]
/lgtm

File: Dockerfile (hunk 1 above)
Comment (ialidzhikov) [reply Dockerfile:25]: It is required to make the build for this PR happy. For more details see https://github.com/gardener/gardener-extension-provider-gcp/pull/69/files#r421264085. After PR, a follow-up cleanup PR such as https://github.com/gardener/gardener-extension-provider-aws/pull/272/files is required.

**Comment by rfranzke** [comment]
@minchaow Is anything preventing the merge?

**Comment by minchaow** [comment]
@rfranzke I am fine with the PR, also approved it already. But I supposed second-opinion probably required before I merge it. :)

**Comment by rfranzke** [comment]
OK, thanks, maybe @kon-angelo can take a second look (I think he recently did a similar migration for other provider extensions).
/invite @kon-angelo

**Review by kon-angelo — APPROVED** [review]

**Pull request merged**